server/src-lib/Hasura/Server/Auth/JWT.hs

```diff
@@ -282,7 +405,7 @@ processJwt_ processAuthZHeader_ jwtCtx headers mUnAuthRole =
       when (requestedRole `notElem` allowedRoles) $
         throw400 AccessDenied "Your requested role is not in allowed roles"
       let finalClaims =
-            Map.delete defaultRoleClaim . Map.delete allowedRolesClaim $ claimsMap
+            Map.delete (sessionVariableToText defaultRoleClaim) . Map.delete (sessionVariableToText allowedRolesClaim) $ claimsMap
```

Comment:
I actually did that, but then I saw the AuthSpec tests required a `J.Object` for the tests, so I had to make it into a `J.Object`. I'll modify the tests and the code to use a `Map SessionVariable J.Value` instead of a `J.Object`

Reply:
I have made the changes